pkg/helm/watcher/controller/helm_watcher.go

```diff
@@ -140,3 +159,82 @@ func (r *HelmWatcherReconciler) reconcileDelete(ctx context.Context, repository
 	// Stop reconciliation as the object is being deleted
 	return ctrl.Result{}, nil
 }
+
+// sendEvent emits a Kubernetes event and forwards the event to notification controller if configured.
+func (r *HelmWatcherReconciler) sendEvent(log logr.Logger, hr *sourcev1.HelmRepository, severity, profileName, version string) {
+	if r.ExternalEventRecorder == nil {
+		return
+	}
+
+	objRef, err := reference.GetReference(r.Scheme, hr)
+	if err != nil {
+		log.Error(err, "unable to get reference")
+		return
+	}
+
+	var meta map[string]string
+	if hr.Status.Artifact.Revision != "" {
+		meta = map[string]string{"revision": hr.Status.Artifact.Revision}
+	}
+
+	if err := r.ExternalEventRecorder.EventInfof(*objRef, meta, severity, "New version available for profile %s with version %s", profileName, version); err != nil {
+		log.Error(err, "unable to send event")
+		return
+	}
+}
+
+// checkForNewVersion uses existing data to determine if there are newer versions in the incoming data
+// compared to what's already stored in the cache. It returns the LATEST version which is greater than
+// the last version that was stored.
+func (r *HelmWatcherReconciler) checkForNewVersion(ctx context.Context, chart *pb.Profile) (string, error) {
+	versions, err := r.Cache.ListAvailableVersionsForProfile(ctx, chart.GetHelmRepository().GetNamespace(), chart.GetHelmRepository().GetName(), chart.Name)
+	if err != nil {
+		return "", err
+	}
+
+	newVersions, err := r.convertStringListToSemanticVersionList(chart.AvailableVersions)
+	if err != nil {
+		return "", err
+	}
+
+	oldVersions, err := r.convertStringListToSemanticVersionList(versions)
+	if err != nil {
+		return "", err
+	}
+
+	sort.SliceStable(newVersions, func(i, j int) bool {
```

Comment:
This could be a local func value...or arguably `func sortVersions(versions []*semver.Version)`

Reply:
Heh, I knew you would say that. :) I was debating to extract it too.